up42/viztools.py

```diff
@@ -308,29 +322,42 @@ def map_results(
                 "You first need to download the results via job.download_results()!"
             )
 
-        # Add features to map.
-        # Some blocks store vector results in an additional geojson file.
-        # pylint: disable=not-an-iterable
-        json_fp = [fp for fp in self.results if fp.endswith(".geojson")]
```

Comment:
@chrieke I could not really understand which specific scenario this line handles. Could you give an example for better understanding!?

Reply:
E.g. the ais blocks have a data.json file with the aoi geometry, and a "ship-detection.geojson" (names differ) file with the actual results geometries. So in that case we want to prioritize the geojson file. Getting the first element is just to get rid of the list.

Reply:
Thanks. I understand it now. Currently,  it handles the specific case (i.e. ship-detection). Plus, as @nekhtiari mentioned about the scenario where a block outputs a `.geojson` (e.g. OSM). I propose we have a rather generic solution for plotting/mapping functionality that handles most (if-not all/unforeseen ones) cases! 
cc: @chrieke @rodrigoalmeida94 @markusuwe